src/coreclr/jit/rationalize.cpp

```diff
@@ -615,6 +615,28 @@ Compiler::fgWalkResult Rationalizer::RewriteNode(GenTree** useEdge, Compiler::Ge
             RewriteIndir(use);
             break;
 
+        case GT_CALL:
+            // In linear order we no longer need to retain the stores in early
+            // args as these have now been sequenced.
+            for (CallArg& arg : node->AsCall()->gtArgs.EarlyArgs())
+            {
+                if (!arg.GetEarlyNode()->IsValue())
+                {
+                    // GTF_LATE_ARG is no longer meaningful.
+                    arg.GetEarlyNode()->gtFlags &= ~GTF_LATE_ARG;
+                    arg.SetEarlyNode(nullptr);
```

Comment:
`GTF_CALL_M_RETBUFFARG_LCLOPT` has a dependency on being able to reach the setup node from the call.

Currently it will work ok because liveness will count the `LCL_VAR|FLD_ADDR` as the definition point, but that itself is only fine because `GTF_CALL_M_RETBUFFARG_LCLOPT` is tied to `lvIsTemp`.

Would be curious to know your thinking on this; my idea was to forbid forcing the buffer arguments to temps.

Reply:
I'm surprised it would ever be evaluated to a temp since it should be invariant. But yes, we should avoid creating a temp for it.

Reply:
> I'm surprised it would ever be evaluated to a temp since it should be invariant

Agree, it is mostly a "model" concern, without `GTF_CALL_M_RETBUFFARG_LCLOPT` forcing things to temps is always legal, with it (+ this change), it isn't.

Reply:
I opened #68469 to stop it looking into the stores and  to assert the contract more heavily.